aidial_adapter_openai/gpt4_multi_modal/download.py

```diff
@@ -23,30 +24,55 @@
 
 def guess_attachment_type(attachment: dict) -> Optional[str]:
     type = attachment.get("type")
-    if type is None:
-        return None
 
-    if "octet-stream" in type:
+    if type is None or "octet-stream" in type:
         # It's an arbitrary binary file. Trying to guess the type from the URL.
         url = attachment.get("url")
         if url is not None:
             url_type = mimetypes.guess_type(url)[0]
             if url_type is not None:
                 return url_type
-        return None
 
     return type
 
 
+class ImageFail(BaseModel):
+    class Config:
+        frozen = True
+
+    name: str
+    message: str
+
+
+async def get_attachment_name(
+    file_storage: Optional[FileStorage], attachment: dict
+) -> str:
+    if "data" in attachment:
+        return attachment.get("title") or "data attachment"
+
+    if "url" in attachment:
+        attachment_link = attachment["url"]
+        if file_storage is not None:
+            return await file_storage.get_human_readable_name(attachment_link)
+        return attachment_link
+
+    return "invalid attachment"
+
+
 async def download_image(
     file_storage: Optional[FileStorage], attachment: dict
-) -> ImageDataURL | str:
+) -> ImageDataURL | ImageFail:
+    name = await get_attachment_name(file_storage, attachment)
+
+    def fail(message: str) -> ImageFail:
+        return ImageFail(name=name, message=message)
+
     try:
         type = guess_attachment_type(attachment)
         if type is None:
-            return "Can't derive media type of the attachment"
+            return fail("can't derive media type of the attachment")
         elif type not in SUPPORTED_IMAGE_TYPES:
-            return f"The attachment isn't one of the supported types: {type}"
+            return fail("the attachment is not one of the supported types")
```

Comment:
Why do you remove this part `types: {type}"`?
It's additional info, that could be helpful

Reply:
It's debatable: first of all, this message is a **user-facing** error message, meaning that the Chat user will see it in a red error box.

Secondly, `type` is a **MIME type**. So arguably this is not a very useful information the Chat user, since it's very likely one doesn't have any notion of MIME types. One would rather familiar with file extensions.

And finally, we print the usage message enumerating the supported file extensions and that should be enough of a help for the user to figure out why the attachment isn't correct.